cmake/CompilerFlags.cmake

```diff
@@ -58,6 +58,12 @@ ELSEIF ( CMAKE_COMPILER_IS_GNUCXX OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang"
       set(LINKER_FLAGS "${LINKER_FLAGS} -fsanitize=undefined -ggdb")
     endif()
 
+    option(ENABLE_COVERAGE "Enable Coverage Reporting in GCC" FALSE)
```

Comment:
Note that we have two different names for options.  ENABLE_FOO and BUILD_BAR.  I guess I can kind of see the logic, build turns on an entirely new target to build.  Enable just alters the compile process in some way.  Do you agree with that?  My first inclination is to always use one or the other, but you first introduced "enable" with the LTO.  It sort of makes sense to me, and I can get on board, provided we agree on the rules.